common/src/lib.rs

```diff
@@ -187,22 +187,38 @@ pub trait DAGBasedLedger {
 pub struct Dispatch {
     pub kind: DispatchKind,
     pub message: Message,
+    pub payload_store: Option<PayloadStore>,
 }
 
 impl Dispatch {
     pub fn new_init(message: Message) -> Self {
         let kind = DispatchKind::Init;
-        Dispatch { message, kind }
+        let payload_store: Option<PayloadStore> = None;
```

Comment:
No need to point the type.
```suggestion
        let payload_store = None;
```